src/tools/sssctl/sssctl_data.c

```diff
@@ -311,3 +314,222 @@ errno_t sssctl_cache_expire(struct sss_cmdline *cmdline,
+
+    for (domain = domain_list;
+         domain;
+         domain = get_next_domain(domain, 0)) {
+        domain_count++;
+    }
+
+    /* allocate output space */
+    domains = talloc_array(ctx, char *, domain_count + 1);
+    if (domains == NULL) {
+        DEBUG(SSSDBG_CRIT_FAILURE,
+              "Could not allocate memory for sections\n");
+        ret = ENOMEM;
+        goto fail;
+    }
+
+    for (domain = domain_list;
+         domain;
+         domain = get_next_domain(domain, 0), i++) {
+        domains[i] = talloc_asprintf(tmp_ctx, "%s", domain->name);
+        if (domains[i] == NULL) {
+            DEBUG(SSSDBG_CRIT_FAILURE, "talloc_asprintf() failed\n");
+            ret = ENOMEM;
+            goto fail;
+        }
+    }
+
+    /* add NULL to the end */
+    domains[i] = NULL;
+
+    *_domains = talloc_steal(ctx, domains);
+
+    return EOK;
+fail:
+    talloc_free(tmp_ctx);
+    return ret;
+}
+
+static errno_t sssctl_cache_index_action(enum sysdb_index_actions action,
+                                         const char **domains,
+                                         const char *attr)
+{
+    errno_t ret;
+    TALLOC_CTX *tmp_ctx = NULL;
+    struct confdb_ctx *confdb = NULL;
+    char *cache;
+    const char **domain;
+    const char **index;
+    const char **indexes = NULL;
+
+    tmp_ctx = talloc_new(NULL);
+    if (tmp_ctx == NULL) {
+        DEBUG(SSSDBG_CRIT_FAILURE, "Failed to allocate the context\n");
+        return ENOMEM;
+    }
+
+    if (domains == NULL) {
+        /* If the user selected no domain, act on all of them */
+        ret = sss_tool_connect_to_confdb(tmp_ctx, &confdb);
+        if (ret != EOK) {
+            DEBUG(SSSDBG_OP_FAILURE,
+                  "Could not connect to configuration database.\n");
+            goto done;
+        }
+
+        ret = get_confdb_domains(tmp_ctx, confdb, discard_const(&domains));
+        if (ret != EOK) {
+            DEBUG(SSSDBG_OP_FAILURE, "Could not list all the domains.\n");
+            goto done;
+        }
+    }
+
+    for (domain = domains; *domain != NULL; domain++) {
+        if (action == SYSDB_IDX_CREATE) {
+            PRINT("Creating cache index for domain %1$s\n", *domain);
+        } else if (action == SYSDB_IDX_DELETE) {
+            PRINT("Deleting cache index for domain %1$s\n", *domain);
+        } else if (action == SYSDB_IDX_LIST) {
+            PRINT("Indexes for domain %1$s:\n", *domain);
+        } else {
+            DEBUG(SSSDBG_CRIT_FAILURE, "Invalid action: %i\n", action);
+            ret = EINVAL;
+            goto done;
+        }
+
+        ret = sysdb_get_db_file(tmp_ctx, NULL, *domain, DB_PATH, &cache, NULL);
+        if (ret != EOK) {
+            DEBUG(SSSDBG_OP_FAILURE, "Failed to get the cache db name\n");
+            goto done;
+        }
+
+        ret = sysdb_manage_index(tmp_ctx, action, cache, attr, &indexes);
+        if (ret != EOK) {
+            goto done;
```

Comment:
It would be nice to print some type of error message here.

Reply:
Here we can only print a generic message. A more specific message is already printed by `sysdb_nmanage_index()`.

Reply:
You are right. I'm closing this issue.